controllers/argocd/statefulset.go

```diff
@@ -441,13 +442,61 @@ func getArgoControllerContainerEnv(cr *argoprojv1a1.ArgoCD) []corev1.EnvVar {
 	return env
 }
 
-func (r *ReconcileArgoCD) reconcileApplicationControllerStatefulSet(cr *argoprojv1a1.ArgoCD, useTLSForRedis bool) error {
+func (r *ReconcileArgoCD) getArgoControllerReplicaCount(cr *argoprojv1a1.ArgoCD) int32 {
 	var replicas int32 = common.ArgocdApplicationControllerDefaultReplicas
+	var minShards int32 = cr.Spec.Controller.Sharding.MinShards
+	var maxShards int32 = cr.Spec.Controller.Sharding.MaxShards
+
+	if cr.Spec.Controller.Sharding.DynamicScalingEnabled {
+		if minShards < 1 {
```

Comment:
Is this check really needed ? As the `ValidatingWebHook` should restrict such invalid configuration.

Reply:
I'd not agree with this conclusion. The validation webhook should never replace sanity checks within the code. Imagine a validation webhook analogue to JavaScript code validating user input in a web UI. You still want to make sure that data is valid in the backend, and not rely on the frontend validation.